src/opentracing-c/tracer.c

```diff
@@ -0,0 +1,219 @@
+#include <opentracing-c/tracer.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+static void null_destroy(opentracing_destructible* destructible)
+{
+    (void) destructible;
+}
+
+#define NULL_DESTRUCTIBLE_INIT \
+    {                          \
+        &null_destroy          \
+    }
+
+static void null_foreach_baggage_item(opentracing_span_context* span_context,
+                                      opentracing_bool (*f)(void*,
+                                                            const char*,
+                                                            const char*),
+                                      void* arg)
+{
+    (void) span_context;
+    (void) f;
+    (void) arg;
+}
+
+#define NULL_SPAN_CONTEXT_INIT                             \
+    {                                                      \
+        NULL_DESTRUCTIBLE_INIT, &null_foreach_baggage_item \
+    }
+
+typedef struct null_span {
+    opentracing_span base;
+    opentracing_span_context null_span_context;
+    opentracing_tracer* null_tracer;
+} null_span;
```

Comment:
Any reason you're using the name null instead of noop which is what it's called in the other APIs (e.g. [go](https://github.com/opentracing/opentracing-go/blob/master/noop.go), [java](https://github.com/opentracing/opentracing-java/tree/master/opentracing-noop))

Reply:
Really? I don't have a strong preference, but I find no-op hard to spell in code (is it `NoOp` or `Noop`?). Null has easy breaking and brings the [null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) to mind.

Reply:
That might be arguments for null if this API was considered in insolation; but given that all the other tracing APIs use noop, I think it's going to be confusing to have the C-API use a different naming scheme.